cassandra-4/src/main/java/io/debezium/connector/cassandra/Cassandra4CommitLogProcessor.java

```diff
@@ -96,23 +94,25 @@ public void stop() {
         super.stop();
     }
 
-    protected synchronized static void removeProcessing(CommitLogProcessingCallable callable) {
+    protected synchronized static void removeProcessing(AbstractCassandra4CommitLogParser parser) {
         submittedProcessings.stream()
-                .filter(p -> p.getFirst() == callable)
+                .filter(p -> p.getFirst() == parser)
                 .findFirst()
                 .map(submittedProcessings::remove);
     }
 
-    final static Set<Pair<CommitLogProcessingCallable, Future<ProcessingResult>>> submittedProcessings = ConcurrentHashMap.newKeySet();
-
     private void submit(Path index) {
-        CommitLogProcessingCallable callable = new CommitLogProcessingCallable(new LogicalCommitLog(index.toFile()),
-                queues,
-                metrics,
-                Cassandra4CommitLogProcessor.this.context);
+        final AbstractCassandra4CommitLogParser parser;
 
-        Future<ProcessingResult> future = executorService.submit(callable);
-        submittedProcessings.add(new Pair<>(callable, future));
+        if (context.getCassandraConnectorConfig().isCommitLogRealTimeProcessingEnabled()) {
+            parser = new Cassandra4CommitLogRealTimeParser(new LogicalCommitLog(index.toFile()), queues, metrics, this.context);
+        }
+        else {
+            parser = new Cassandra4CommitLogBatchParser(new LogicalCommitLog(index.toFile()), queues, metrics, this.context);
+        }
+
+        Future<ProcessingResult> future = executorService.submit(() -> parser.process());
```

Comment:
`() -> parser.process()` might be replaced with method reference like `.submit(parser::proces)`.